### PR TITLE
Lint documentation examples

### PR DESCRIPTION
**Pull request by brylie**
The documentation example(s) use unconventional formatting. Using `pylint` and `black` to improve formatting.

## Includes
- [ ] Code changes
- [ ] Tests
- [X] Documentation

### CODE REVIEW

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py [hunk 1]
```diff
@@ -25,28 +25,7 @@ def __init__(self):
 
         # GUI
         box = gui.widgetBox(self.controlArea, "Info")
-        self.infoa = gui.widgetLabel(box, 'No data on input yet, waiting to get something.')
-        self.infob = gui.widgetLabel(box, '')
-# [end-snippet-1]
-
-# [start-snippet-2]
-    @Inputs.data
```
Comment (ajdapretnar) [inline doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: Why is this part removed?

**Comment by ajdapretnar** [comment]
conf.py files are auto-generated from Sphinx, so in the long run it might be futile to change them.

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py (hunk 1 above)
Comment (brylie) [reply doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: I think the indentation was just changed, there shouldn't be any removals.

Comment (brylie) [reply doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: That is strange. I will double check why that might have happened.

File: doc/data-mining-library/source/tutorial/code/data-feature-selection.py [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import Orange
 
 data = Orange.data.Table("iris.tab")
-new_domain = Orange.data.Domain(list(data.domain.attributes[:2]),
-                                data.domain.class_var)
+new_domain = Orange.data.Domain(list(data.domain.attributes[:2]), data.domain.class_var)
```
Comment (ajdapretnar) [inline doc/data-mining-library/source/tutorial/code/data-feature-selection.py]: This is not PEP8-compliant. I'd prefer having lines shorter than 80 chars.

File: doc/data-mining-library/source/tutorial/code/regression-other.py [hunk 3]
```diff
@@ -17,6 +17,8 @@
 print("y   ", " ".join("%5s" % l.name for l in regressors))
 
 for d in test:
-    print(("{:<5}" + " {:5.1f}"*len(regressors)).format(
-        d.get_class(),
-        *(r(d) for r in regressors)))
+    print(
+        ("{:<5}" + " {:5.1f}" * len(regressors)).format(
+            d.get_class(), *(r(d) for r in regressors)
```
Comment (ajdapretnar) [inline doc/data-mining-library/source/tutorial/code/regression-other.py]: This is broken in a strange place.

File: doc/data-mining-library/source/tutorial/code/data-feature-selection.py (hunk 2 above)
Comment (brylie) [reply doc/data-mining-library/source/tutorial/code/data-feature-selection.py]: I guess Black defaults to 88 character length:
https://github.com/psf/black#line-length

Where do you recommend that I add the configuration, so that Black will respect the 80 character length limit?

Comment (ajdapretnar) [reply doc/data-mining-library/source/tutorial/code/data-feature-selection.py]: Some reasoning: https://www.reddit.com/r/Python/comments/9uxo9f/do_you_use_black_with_its_default_line_length_of/

Apparently you can set it with config override `--line-length=79`. I am not using Black though, so haven't tried it myself.

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py (hunk 1 above)
Comment (brylie) [reply doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: OK, I have restored the lines. Sorry about the misstep.

File: doc/data-mining-library/source/tutorial/code/data-feature-selection.py (hunk 2 above)
Comment (brylie) [reply doc/data-mining-library/source/tutorial/code/data-feature-selection.py]: OK, fixed.

File: doc/development/source/orange-demo/orangedemo/OWLearningCurveA.py [hunk 4]
```diff
@@ -15,17 +15,18 @@
 
 class OWLearningCurveA(OWWidget):
     name = "Learning Curve (A)"
-    description = ("Takes a dataset and a set of learners and shows a "
-                   "learning curve in a table")
+    description = (
+        "Takes a dataset and a set of learners and shows a " "learning curve in a table"
```
Comment (ajdapretnar) [inline doc/development/source/orange-demo/orangedemo/OWLearningCurveA.py]: `shows a " "learning curve`
I think this is not formatted correctly.

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py [hunk 5]
```diff
@@ -25,28 +22,53 @@ def __init__(self):
...
         # GUI
         box = gui.widgetBox(self.controlArea, "Info")
-        self.infoa = gui.widgetLabel(box, 'No data on input yet, waiting to get something.')
+        self.infoa = gui.widgetLabel(
+            box, 'No data on input yet, waiting to get something.')
         self.infob = gui.widgetLabel(box, '')
 # [end-snippet-1]
 
 # [start-snippet-2]
-    @Inputs.data
```
Comment (ajdapretnar) [inline doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: Do not remove the decorator. This is required to set the input data function.

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py [hunk 6]
```diff
@@ -1,22 +1,19 @@
...
     icon = "icons/DataSamplerA.svg"
     priority = 10
 
-    class Inputs:
-        data = Input("Data", Orange.data.Table)
-
-    class Outputs:
-        sample = Output("Sampled Data", Orange.data.Table)
+    inputs = [("Data", Orange.data.Table, "set_data")]
+    outputs = [("Sampled Data", Orange.data.Table)]
```
Comment (ajdapretnar) [inline doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: The above inputs/output are correct. Your version is the deprecated one.

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py [hunk 7]
```diff
@@ -25,28 +22,53 @@ def __init__(self):
 
         # GUI
         box = gui.widgetBox(self.controlArea, "Info")
-        self.infoa = gui.widgetLabel(box, 'No data on input yet, waiting to get something.')
+        self.infoa = gui.widgetLabel(
+            box, 'No data on input yet, waiting to get something.')
```
Comment (ajdapretnar) [inline doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: In this case, should the label not also be enveloped in ""?

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py [hunk 8]
```diff
@@ -25,28 +22,53 @@ def __init__(self):
...
     def set_data(self, dataset):
         if dataset is not None:
-            self.infoa.setText('%d instances in input dataset' % len(dataset))
+            self.infoa.setText('%d instances in input data set' % len(dataset))
             indices = numpy.random.permutation(len(dataset))
             indices = indices[:int(numpy.ceil(len(dataset) * 0.1))]
             sample = dataset[indices]
             self.infob.setText('%d sampled instances' % len(sample))
-            self.Outputs.sample.send(sample)
+            self.send("Sampled Data", sample)
```
Comment (ajdapretnar) [inline doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: Same as above: the former version was correct.

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py [hunk 9]
```diff
@@ -25,28 +22,53 @@ def __init__(self):
...
         else:
-            self.infoa.setText('No data on input yet, waiting to get something.')
+            self.infoa.setText(
+                'No data on input yet, waiting to get something.')
             self.infob.setText('')
-            self.Outputs.sample.send("Sampled Data")
+            self.send("Sampled Data", None)
```
Comment (ajdapretnar) [inline doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: Same as above: the former version was correct.

File: doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py [hunk 10]
```diff
@@ -25,28 +22,53 @@ def __init__(self):
...
 # [end-snippet-2]
 
-
 # [start-snippet-3]
+
+
+def main(argv=sys.argv):
+    from PyQt4.QtGui import QApplication
```
Comment (ajdapretnar) [inline doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py]: I would avoid using PyQt4. Rather AnyQt or PyQt5.
